web/components/filter.tsx

```diff
@@ -126,6 +126,56 @@ export function FilterList({
     };
   }, []);
 
+  useEffect(() => {
+    // TODO: why is this called after each setSelectionOption
+    // TODO: up / down also moves cursor inside MiniEditor to begin / end
+    function onKeyDown(e: KeyboardEvent) : boolean {
+      const nOptions = matchingOptions.length;
+      if (nOptions === 0) {
+        return false;
+      }
+      let newSelected : number | undefined;
+      switch (e.key) {
+        case "ArrowUp":
+          newSelected = selectedOption - 1;
+          break;
+        case "ArrowDown":
+          newSelected = selectedOption + 1;
+          break;
+        case "PageUp":
+          newSelected = selectedOption - 5;
+          break;
+        case "PageDown":
+          newSelected = selectedOption + 5;
+          break;
+        case "Home":
+          newSelected = 0;
+          break;
+        case "End":
+          newSelected = nOptions - 1;
+          break;
+      }
+      if (newSelected === undefined || nOptions < 1) {
+        return true;
+      }
+      if (newSelected < 0) {
+        newSelected = 0;
+      }
+      if (newSelected >= nOptions) {
+        newSelected = nOptions - 1;
+      }
+      setSelectionOption(newSelected);
+      return true;
+    }
+
+    document.addEventListener("keydown", onKeyDown);
+
+    return () => {
+      document.removeEventListener("keydown", onKeyDown);
+    };
+
+  }, [selectedOption, matchingOptions]);
```

Comment:
```suggestion
  }, []);
```

to do this just once?